modules/Hoot/ui/modals/ImportMultiDatasets.js

```diff
@@ -387,41 +402,73 @@ export default class ImportMultiDatasets {
                         }
 
                         Hoot.message.alert( { message, type, keepOpen } );
-                    })
-                    .finally( () => {
-                        this.container.remove();
-                        Hoot.events.emit( 'modal-closed' );
-                    } );
-            } );
-        } );
-
-        this.processRequest = this.allProgress( proms, ( n, fileName ) => {
-                this.progressBar.property( 'value', n );
-                this.fileListInput
-                    .select( `option[value="${fileName}"]` )
-                    .classed( 'import-success', true );
-            } )
-            .then( () => Hoot.message.alert( {
-                message: 'All datasets successfully imported',
-                type: 'success'
-            } ) )
-            .then( () => Hoot.folders.refreshAll() )
-            .then( () => Hoot.events.emit( 'render-dataset-table' ) );
-    }
-
-    allProgress( proms, cb ) {
-        let n = 0;
+                    });
+            } )];
+        }
+        else {
+            proms = fileNames.map( (name, index) => {
+                return new Promise( resolve => {
+                    let importFiles = _filter( this.fileIngest.node().files, file => {
+                        let fName = file.name.substring( 0, file.name.length - 4 );
+
+                        if ( file.name.toLowerCase().indexOf( '.shp.xml' ) > -1 ) {
+                            fName = file.name.substring( 0, file.name.length - 8 );
+                        }
+                        if ( file.name.indexOf( '.geojson' ) > -1 ) {
+                            fName = file.name.toLowerCase().substring( 0, file.name.length - 8 );
+                        }
 
-        cb( 0 );
+                        return fName === name;
+                    } );
 
-        proms.forEach( p => {
-            p.then( fileName => {
-                n++;
-                cb( n, fileName );
+                    let params = {
+                        NONE_TRANSLATION: translation.NONE === 'true',
+                        TRANSLATION: translationName,
+                        INPUT_TYPE: importType.value,
+                        INPUT_NAME: name,
+                        formData: this.getFormData( importFiles ),
+                        folderId
+                    };
+
+                    Hoot.api.uploadDataset( params )
+                        .then( resp => Hoot.api.statusInterval( resp.data[ 0 ].jobid ) )
+                        .then( () => {
+                            this.progressBar.property( 'value', index + 1 );
+                            this.fileListInput
+                                .select( `option[value="${name}"]` )
+                                .classed( 'import-success', true );
+                        } )
+                        .then( () => resolve( name ) )
```

Comment:
Not sure resolve is needed any longer as you've moved the callback code (updating progress bar and filename highlighting) directly into the promise.

However, I think something can be done to make this section of code more DRY.  The async method in both the asSingle and normal multiple upload are almost identical except for the filtering of files into each set to upload and the import-success selector.  I think we could just set a variable to skip the filtering of importFiles and the `[value=]' css selector